server/task/task-distributed/src/main/scala/org/apache/james/task/eventsourcing/cassandra/CassandraTaskExecutionDetailsProjectionModule.scala

```diff
@@ -44,19 +45,20 @@ object CassandraTaskExecutionDetailsProjectionModule {
 
   val MODULE: CassandraModule = CassandraModule.table(CassandraTaskExecutionDetailsProjectionTable.TABLE_NAME)
     .comment("Projection of TaskExecutionDetails used by the distributed task manager")
-    .options(options => options.withCaching(true, SchemaBuilder.RowsPerPartition.NONE))
+    .options(options => options.withCaching(true, SchemaBuilder.RowsPerPartition.NONE)
+      .withClusteringOrder(CassandraTaskExecutionDetailsProjectionTable.SUBMITTED_DATE, ClusteringOrder.ASC))
     .statement(statement => types => statement
       .withPartitionKey(CassandraTaskExecutionDetailsProjectionTable.TASK_ID, DataTypes.UUID)
+      .withClusteringColumn(CassandraTaskExecutionDetailsProjectionTable.SUBMITTED_DATE, DataTypes.TIMESTAMP)
       .withColumn(CassandraTaskExecutionDetailsProjectionTable.ADDITIONAL_INFORMATION, DataTypes.TEXT)
       .withColumn(CassandraTaskExecutionDetailsProjectionTable.TYPE, DataTypes.TEXT)
       .withColumn(CassandraTaskExecutionDetailsProjectionTable.STATUS, DataTypes.TEXT)
-      .withColumn(CassandraTaskExecutionDetailsProjectionTable.SUBMITTED_DATE, types.getDefinedUserType(CassandraZonedDateTimeModule.ZONED_DATE_TIME))
       .withColumn(CassandraTaskExecutionDetailsProjectionTable.SUBMITTED_NODE, DataTypes.TEXT)
-      .withColumn(CassandraTaskExecutionDetailsProjectionTable.STARTED_DATE, types.getDefinedUserType(CassandraZonedDateTimeModule.ZONED_DATE_TIME))
+      .withColumn(CassandraTaskExecutionDetailsProjectionTable.STARTED_DATE, DataTypes.TIMESTAMP)
       .withColumn(CassandraTaskExecutionDetailsProjectionTable.RAN_NODE, DataTypes.TEXT)
-      .withColumn(CassandraTaskExecutionDetailsProjectionTable.COMPLETED_DATE, types.getDefinedUserType(CassandraZonedDateTimeModule.ZONED_DATE_TIME))
-      .withColumn(CassandraTaskExecutionDetailsProjectionTable.CANCELED_DATE, types.getDefinedUserType(CassandraZonedDateTimeModule.ZONED_DATE_TIME))
+      .withColumn(CassandraTaskExecutionDetailsProjectionTable.COMPLETED_DATE, DataTypes.TIMESTAMP)
+      .withColumn(CassandraTaskExecutionDetailsProjectionTable.CANCELED_DATE, DataTypes.TIMESTAMP)
       .withColumn(CassandraTaskExecutionDetailsProjectionTable.CANCEL_REQUESTED_NODE, DataTypes.TEXT)
-      .withColumn(CassandraTaskExecutionDetailsProjectionTable.FAILED_DATE, types.getDefinedUserType(CassandraZonedDateTimeModule.ZONED_DATE_TIME)))
+      .withColumn(CassandraTaskExecutionDetailsProjectionTable.FAILED_DATE, DataTypes.TIMESTAMP))
```

Comment:
Changing a Cassandra module is a hard red line!

What's the migration strategy here? Can we avoid such a breaking change?

IMO we would be better of not toucing the data layer in this pull request.

Reply:
My idea is we can filter "olderThan" at db layer. But It will be hard with UDTType when orderBy.

Reply:
I understand your idea but it have serious consequences on existing data. 

Existing data needs considerations, your changes would break existing instalations. 

We need to be careful with data structure changes, do them only if necessary and come up with a migration strategy.

Here my feeling is that we don't need it. Cassandra is OK reading 100 thousands entries from time to time (would already be extreme for this use case, we 'd rather be with a rough estimate more on the thousand entry range). Apps are ok filtering such data. Early optimisation is the root of all evil. Let's not try to optimize this just yet, until we consider that optimisation is truely needed.

Reply:
Thank for your explain